File: backend/app/schemas/notification.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class NotificationRead(BaseModel):
    id: int
    task_id: int | None
    type: str
    title: str
    body: str
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class UnreadCount(BaseModel):
    count: int

File: frontend/src/types/notification.ts
export type NotificationType = 'reminder_3d' | 'reminder_1d' | 'overdue'

export interface NotificationItem {
  id: number
  task_id: number | null
  type: NotificationType | string
  title: string
  body: string
  is_read: boolean
  link_url: string | null
  created_at: string
}

export interface UnreadCount {
  count: number
}

export interface WsNotification {
  type: 'notification'
  id: number
  notification_type: string
  title: string
  body: string
  task_id: number | null
  link_url: string | null
}
